src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/NoviflowSpecificFeature.java

```diff
@@ -23,15 +23,28 @@
 abstract class NoviflowSpecificFeature extends AbstractFeature {
     public static final String NOVIFLOW_MANUFACTURER_SUFFIX = "noviflow";
 
-    boolean isNoviSwitch(IOFSwitch sw) {
-        Optional<SwitchDescription> description = Optional.ofNullable(sw.getSwitchDescription());
-        return description
-                .map(SwitchDescription::getManufacturerDescription)
+    static boolean isNoviSwitch(IOFSwitch sw) {
+        if (sw == null || sw.getSwitchDescription() == null) {
+            return false;
+        }
+
+        SwitchDescription description = sw.getSwitchDescription();
+
+        if (E_SWITCH_MANUFACTURER_DESCRIPTION.equalsIgnoreCase(description.getManufacturerDescription())) {
+            return true;
+        }
+
+        if (E_SWITCH_HARDWARE_DESCRIPTION_REGEX.matcher(
+                Optional.ofNullable(description.getHardwareDescription()).orElse("")).matches()) {
+            return true;
+        }
+
+        return Optional.ofNullable(description.getManufacturerDescription())
                 .map(String::toLowerCase)
                 .orElse("").contains(NOVIFLOW_MANUFACTURER_SUFFIX);
     }
 
-    boolean is100GbHw(IOFSwitch sw) {
+    static boolean is100GbHw(IOFSwitch sw) {
```

Comment:
let's rename it to something like isWBSeries

Reply:
fixed